components/director/pkg/http/request_test.go

```diff
@@ -0,0 +1,132 @@
+package http_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	httputil "github.com/kyma-incubator/compass/components/director/pkg/http"
+	"github.com/stretchr/testify/require"
+)
+
+const testURL = "http://localhost:8000"
+
+var expectedResp = &http.Response{
+	StatusCode: http.StatusOK,
+	Body:       nil,
+}
+
+type RoundTripFunc func(req *http.Request) *http.Response
+
+func (f RoundTripFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req), nil
+}
```

Comment:
It would be better to check if `f` returns nil and return error from the `RoundTrip` then the `TestRequestWithoutCredentials_FailedRequest` and all the other negative tests will return the actual error.